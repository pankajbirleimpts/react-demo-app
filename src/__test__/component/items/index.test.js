import React from "react";
import { shallow } from "enzyme";
import ItemList from "../../../components/items/index";
import { storeFactory } from "../../test/TestUtils";

const setup = props => {
  const store = storeFactory(props);
  const wrapper = shallow(<ItemList store={store} />)
    .dive()
    .dive();
  return wrapper;
};

describe("ItemList component", () => {
  const props = {
    item: {
      isLoading: false,
      allItems: [
        {
          id: 1,
          amount: 15,
          category: "Roti",
          description: "This is tawa rohi",
          isActive: "Active",
          itemName: "Tawa Roti"
        },
        {
          amount: 15,
          category: "Dal",
          description: "Tal tadka",
          isActive: "Active",
          itemName: "Dal tadka"
        }
      ]
    }
  };
  test("Snapshot of ItemList component", () => {
    const wrapper = setup(props);
    expect(wrapper).toMatchSnapshot();
  });

  test("Check the isLoading in props", () => {
    const wrapper = setup(props);
    const componentProps = wrapper.instance().props;
    expect(componentProps.item.isLoading).toBe(props.item.isLoading);
  });

  test("Check the `getAllItems` action in props", () => {
    const wrapper = setup(props);
    const componentProps = wrapper.instance().props;
    expect(componentProps.getAllItems).toBeInstanceOf(Function);
  });
});
