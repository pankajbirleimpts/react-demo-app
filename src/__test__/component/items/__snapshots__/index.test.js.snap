// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList component Snapshot of ItemList component 1`] = `
<Grid>
  <Loading
    isLoading={false}
  />
  <GridRow>
    <GridColumn
      textAlign="left"
      width={8}
    >
      <h3>
        Manage Items
      </h3>
    </GridColumn>
    <GridColumn
      textAlign="right"
      width={8}
    >
      <Link
        className="ui button primary"
        to="/add-item"
      >
        + Add item
      </Link>
    </GridColumn>
  </GridRow>
  <GridRow
    centered={true}
  >
    <Connect(UnconnectedCustomTable)
      columns={
        Array [
          Object {
            "keyName": "itemName",
            "label": "Item Name",
            "searchable": true,
            "sort": true,
            "sortable": true,
          },
          Object {
            "keyName": "category",
            "label": "Category",
            "searchable": true,
            "sort": null,
            "sortable": true,
          },
          Object {
            "keyName": "isActive",
            "label": "Status",
            "searchable": true,
            "sort": null,
            "sortable": true,
          },
          Object {
            "keyName": "amount",
            "label": "Amount",
            "searchable": true,
            "sort": null,
            "sortable": true,
          },
          Object {
            "keyName": "description",
            "label": "Description",
            "searchable": true,
            "sort": null,
            "sortable": true,
          },
        ]
      }
      module="ITEMS"
      tableData={
        Array [
          Object {
            "amount": 15,
            "category": "Roti",
            "description": "This is tawa rohi",
            "id": 1,
            "isActive": "Active",
            "itemName": "Tawa Roti",
          },
          Object {
            "amount": 15,
            "category": "Dal",
            "description": "Tal tadka",
            "isActive": "Active",
            "itemName": "Dal tadka",
          },
        ]
      }
    />
  </GridRow>
</Grid>
`;
